packages/bff/src/model/api/apiConverter.ts

```diff
@@ -168,3 +170,67 @@ export function toTenantWithOnlyAttributes(
     attributes: tenant.attributes.map(toTenantAttribute).flat(),
   };
 }
+
+export function toApiDeclaredTenantAttribute(
+  attribute: tenantApi.TenantAttribute,
+  registryAttributeMap: Map<string, attributeRegistryApi.Attribute>
+): bffApi.DeclaredTenantAttribute | null {
```

Comment:
this methods should be used to convert something, I'm not expect that could be return `null` or `undefined` , I prefer to leave null check to invoker and using converts method only on concrete object, furthermore we want discourage `null` usages.

Reply:
I want to skip null-checks in all methods. I prefer to avoid handling null values after transforming to the API response type (these methods could be reused by others), so I would rather write a function that accepts `attribute.{DECLARED/CERTIFIED/VERIFIED}` only when defined. However, in this case, `registryAttributeMap.get` may not find the item 😞, so it's a bit different...

Anyway, for me the only change required is returning `undefined` instead of `null`. Secondly, verify if it's possible to apply some changes in [enhanceTenantAttributes](https://github.com/pagopa/interop-be-monorepo/pull/722/files/0c5914771651258c5f7323bb6252526afc591ffa#diff-d0defbd6f7a8d5725ba5941c4b26796fe8e04f132011fdd14ebf27886d6fa50cR402) to invoke the mapping function only when `registryAttributeMap` contains the target attribute and the target attribute is defined. Otherwise, we can leave it as is for the moment.

Reply:
For now, i moved all the enhanceTenant logic and helpers in `tenantService.ts`, and changed the "converters" return type to `T | undefined`  instaed of `T | null`. I moved the converters away from `apiConverter.ts` because as you said converters should always return a T value directly.

I personally didn't find a way to make this more generic so i still have one function for each type of attribute. I think it may be more readable like this instaed of making a single complicated generic function for all three.